addons/sale_pdf_quote_builder/static/src/js/binary/binary_field.js

```diff
@@ -0,0 +1,30 @@
+import { BinaryField } from "../../../../../web/static/src/views/fields/binary/binary_field";
+import { _t } from "@web/core/l10n/translation";
+import { useService } from "@web/core/utils/hooks";
+import { patch } from "@web/core/utils/patch";
+
+patch(BinaryField.prototype, {
+    setup() {
+        this.notification = useService('notification');
+    },
+
+    update({ data, name }) {
+        const { fileNameField, record } = this.props;
+        const changes = { [this.props.name]: data || false };
+        if (fileNameField in record.fields && record.data[fileNameField] !== name) {
+            if (!name.endsWith('.pdf')) {
+                this.notification.add(
+                    _t(`Oops! '%(fileName)s' didn’t upload since this format isn’t allowed.`, {
+                        fileName: name,
+                    }),
+                    {
+                        type: "danger",
+                    }
+                );
+                return false;
+            }
+            changes[fileNameField] = name || '';
+        }
+        return this.props.record.update(changes);
+    },
+});
```

Comment:
```suggestion

patch(FileUploader.prototype, {
    setup() {
        super.setup();
        this.notification = useService('notification');
    },

    
    async onFileChange(ev) {
        const files = [...ev.target.files].filter(file => this.validFileType(file));
        if (!files.length) {
            return;
        }
        return super.onFileChange(...arguments);
    },

    /**
     * The `allowedMIMETypes` can restrict the file types users are guided to select. However,
     * the `accept` attribute doesn't enforce strict validation; it only suggests file types for
     * browsers.
     *
     * @param {File} file
     * @returns Whether the upload file's type is in the whitelist (`allowedMIMETypes`).
     */
    validFileType(file) {
        const allowedMIMETypes = "application/pdf"
        if (allowedMIMETypes && !allowedMIMETypes.includes(file.type)) {
            this.notification.add(
                _t(`Oops! '%(fileName)s' didn’t upload since its format isn’t allowed.`, {
                    fileName: file.name,
                }),
                {
                    type: "danger",
                }
            );
            return false;
        }
        return true;
    }
});

```

With your changes, we were unable to identify the file type. We should validate a file by its type, not by its name ending with '.pdf'. Users may rename the extension.

Also retarget this branch into saas-17.4 and adapt file name and path with new chages.